p2p/ipld/read.go

```diff
@@ -269,29 +268,7 @@ func leafPath(index, total uint32) ([]string, error) {
 	return path, nil
 }
 
-// nextPowerOf2 returns the next lowest power of 2 unless the input is a power
-// of two, in which case it returns the input
-func nextPowerOf2(v uint32) uint32 {
-	if v == 1 {
-		return 1
-	}
-	// keep track of the input
-	i := v
-
-	// find the next highest power using bit mashing
-	v--
-	v |= v >> 1
-	v |= v >> 2
-	v |= v >> 4
-	v |= v >> 8
-	v |= v >> 16
-	v++
-
-	// check if the input was the next highest power
-	if i == v {
-		return v
-	}
-
-	// return the next lowest power
-	return v / 2
+// isPowerOf2 returns checks if a given number is a power of two
+func isPowerOf2(v uint32) bool {
+	return math.Ceil(math.Log2(float64(v))) == math.Floor(math.Log2(float64(v)))
```

Comment:
Neat! I'm sure there is a way to do this without floats but let's think about this later. The above is certainly clearer than before.